examples/xr/augmented-reality-basic.html

```diff
@@ -0,0 +1,170 @@
+            }
+        </style>
+    </head>
+
+    <body>
+        <canvas id="application-canvas"></canvas>
+        <div>
+            <p class="message"></p>
+        </div>
+        <script>
+            // draw some axes
+            var drawAxes = function (pos, scale) {
+                var color = new pc.Color(1,0,0);
+
+                var axis = new pc.Vec3();
+                var end = new pc.Vec3().copy(pos).add(axis.set(scale,0,0));
+
+                app.renderLine(pos, end, color);
+
+                color.set(0, 1, 0);
+                end.sub(axis.set(scale,0,0)).add(axis.set(0,scale,0));
+                app.renderLine(pos, end, color);
+
+                color.set(0, 0, 1);
+                end.sub(axis.set(0,scale,0)).add(axis.set(0,0,scale));
+                app.renderLine(pos, end, color);
+            }
+        </script>
+
+
+        <script>
+            var message = function (msg) {
+                var el = document.querySelector('.message');
+                el.textContent = msg;
+            }
+
+            var canvas = document.getElementById('application-canvas');
+            var app = new pc.Application(canvas, {
+                mouse: new pc.Mouse(canvas),
+                touch: new pc.TouchDevice(canvas),
+                keyboard: new pc.Keyboard(window)
+            });
+            app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+            app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+            window.addEventListener("resize", function () {
+                app.resizeCanvas(canvas.width, canvas.height);
+            });
+
+            // use device pixel ratio
+            app.graphicsDevice.maxPixelRatio = window.devicePixelRatio;
+
+            app.start();
+
+            // create camera
+            var c = new pc.Entity();
+            c.addComponent('camera', {
+                clearColor: new pc.Color(0, 0, 0, 0),
+                farClip: 10000
+            });
+            app.root.addChild(c);
+
+            var l = new pc.Entity();
+            l.addComponent("light", {
+                type: "spot",
+                range: 30
+            });
+            l.translate(0,10,0);
+            app.root.addChild(l);
+
+
+            var createCube = function(x,y,z) {
+                var cube = new pc.Entity();
+                cube.addComponent("model", {
+                    type: "box",
+                });
+                cube.setLocalScale(.5, .5, .5);
+                cube.translate(x * .5, y, z * .5);
+                app.root.addChild(cube);
+            };
+
+            // create a grid of cubes
+            var SIZE = 4;
+            for (var x = 0; x < SIZE; x++) {
+                for (var y = 0; y < SIZE; y++) {
+                    createCube(2*x - SIZE, -1, 2*y - SIZE);
+                }
+            }
+
+            if (app.xr.supported) {
+                var activate = function () {
+                    if (app.xr.isAvailable(pc.XR_TYPE_IMMERSIVE_AR)) {
```

Comment:
So you have `app.xr.supported` and then `app.xr.isAvailable(pc.XR_TYPE_IMMERSIVE_AR)`. For consistency, and better readability, what about `app.xr.supports(pc.XRTYPE_AR)`

Reply:
Support is static property, that is set before page is loaded. Which defines if platform supports WebXR API at all.
And "availability" - is what type of session currently is available, which can change during page life based on what devices are plugged in, etc.
I've split them intentionally, so not to confuse support with availability.

Support - cannot change, only browser upgrade can.
Availability - can change dynamically during page life.